python/tchannel/tornado/util.py

```diff
@@ -0,0 +1,43 @@
+from __future__ import absolute_import
+
+import tornado
+import tornado.gen
+from ..exceptions import TChannelException
+
+
+@tornado.gen.coroutine
+def get_arg(context, index):
+    if index < len(context.argstreams):
+        arg = ""
+        try:
+            chunk = yield context.argstreams[index].read()
+            while chunk:
+                arg += chunk
+                chunk = yield context.argstreams[index].read()
+        finally:
+            raise tornado.gen.Return(arg)
```

Comment:
When returning values in tornado, don't do it from finally blocks.

If any of the previous code threw an exception, this will cause that exception to be completely ignored and always return a response (because of how `raise` works)